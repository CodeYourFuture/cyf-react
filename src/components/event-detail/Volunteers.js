// @flow
import React from 'react';
import styled from 'react-emotion';
import DisplayPicture from './DisplayPicture';
import type { User as UserType } from './types';

const Container = styled('div')`
  display: flex;
  padding-top: 10px;
  padding-bottom: 10px;
`;

export default ({ users }: { users: UserType[] }) => (
  <Container>
    {users.map(user => <DisplayPicture key={user.user_id} {...user} />)}
  </Container>
);
