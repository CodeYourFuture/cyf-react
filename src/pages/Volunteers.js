import React from 'react';
import { Link } from 'react-router-dom';

import { TopSection } from '../components/TopSection';
import { BottomSection } from '../components/BottomSection';

export const Volunteers = () => {
  const volunteersText = (
    <div>
      <p>
        Welcome to a community where your knowledge and experience will help to
        transform the lives of refugees in the UK.
      </p>
      <p>
        We are looking for experienced web developers (HTML/CSS, JavaScript,
        Node, Angular, React, Database) to participate in our classes.
      </p>
      <p>
        We are also looking for technology managers and entrepreneurs to give
        workshops on modern software practices and methodologies.
      </p>

      <div className="section-bottom-link">
        <Link
          className="big-link-3 btn"
          to="/apply/mentor"
          target="_blank"
          rel="noopener noreferrer"
        >
          Become a Mentor
        </Link>
      </div>

      <p>
        Or you can become a volunteer and help us support the expansion of the
        organisation. No matter where you are in the UK, with your help weâ€™ll
        create a local program to serve local refugee communities.
      </p>

      <BottomSection link="/apply/volunteer" linkText="Become a Volunteer" />
    </div>
  );

  return (
    <div>
      <TopSection title="Join Us" content={volunteersText} />
      <div className="mentors-hero-wrap" />
    </div>
  );
};
