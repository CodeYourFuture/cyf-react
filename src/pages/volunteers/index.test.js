import React from 'react';
import { shallow } from 'enzyme';
import Volunteers from './';

describe('Volunteers Page', () => {
  it('renders correctly', () => {
    expect(shallow(<Volunteers />)).toMatchSnapshot();
  });
});
