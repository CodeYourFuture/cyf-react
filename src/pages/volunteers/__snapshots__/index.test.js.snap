// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Volunteers Page renders correctly 1`] = `
<div>
  <Styled(div)>
    <SectionHeading>
      <h2>
        Join Us
      </h2>
    </SectionHeading>
    <Divider />
    <SectionBody>
      <p>
        Welcome to a community where your knowledge and expertise will help to transform the lives of disadvantaged and underprivileged individuals, refugees and asylum seekers.
      </p>
      <p>
        A young nonprofit is a lot like a start-up: growing quickly and always looking for versatile people to join. Join us in one of 5 CYF cities worldwide or set up a new CYF chapter to support your local communities. We are looking for:
      </p>
      <ul>
        <li>
          <strong>
            Volunteers from virtually any background
          </strong>
          to help us run and grow our organisation.
        </li>
        <li>
          Professionals from the tech industry to join our
           
          <strong>
            mentorship programme
          </strong>
           and give a student guidance and support.
        </li>
        <li>
          <strong>
            Experienced web developers
          </strong>
           (HTML/CSS, JavaScript, Node, React, Databases) to help teach our classes.
        </li>
      </ul>
    </SectionBody>
    <CYFLink
      path="/apply"
      title="Become a Volunteer"
    />
  </Styled(div)>
  <div>
    <img
      alt="code-your-future-mentors"
      src="mentors.jpg"
    />
  </div>
</div>
`;
